src/decide.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+import "google/protobuf/empty.proto";
+
+package decide;
+
+/* The payload for a requested state change to a component. Component state may
+   be a scalar (int32) or a label (string) */
+message StateChange {
+  string component = 1;
+  oneof state {
+    int32 scalar = 4;
+    string label = 5;
+  }
+}
+
+/* The payload for a requested change to the parameters for a component */
+message ComponentParams {
+  string component = 1;
+  // The documentation for the component must define how the parameter value will be parsed from the string.
+  map<string, string> parameters = 2;
+}
+
+/* This is what gets sent on the wire */
+message Request {
+  string protocol = 1;
+  oneof request {
+    // request change to state of a component. Reply type is ok or error.
+    StateChange state_change = 2;
+    // request reset to default state; value is name of component. Reply type is ok or error
+    string state_reset = 3;
+    // request lock on experiment; value is name of experiment. Reply type is ok or error
+    string lock_expt = 16;
```

Comment:
I think we talked about also including an identifier for the apparatus config in the lock request, so that decide-ctrl can reject experiments that are expecting a different arrangement of components.

Reply:
See next...